### PR TITLE
cache containerd 1.6, add testing channel for apt, and switch to packages.microsoft.com again 

### PR DESCRIPTION
**Pull request by ganeshkumar5699**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
This PR is to fix 2 bugs: 
1) Containerd 1.6 not being cached in the VHD.
2) Containerd 1.6 needing additional apt config to use the "testing" channel. 

The switch to using packages.microsoft.com is from this PR: 
https://github.com/Azure/AgentBaker/pull/1900 

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
This PR switches containerd downloads to use `packages.microsoft.com` with apt (instead of `moby.blob.core.windows.net`). After this VHD is released and in the corresponding RP vendor PR, it should make containerd downloads consistent with the AKS egress docs.
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/manifest.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
         "downloadLocation": "/opt/containerd/downloads",
         "downloadURL": "https://moby.blob.core.windows.net/moby/moby-containerd/${CONTAINERD_VERSION}+azure/bionic/linux_${CPU_ARCH}/moby-containerd_${CONTAINERD_VERSION}+azure-ubuntu18.04u${CONTAINERD_PATCH_VERSION}_${CPU_ARCH}.deb",
         "versions": [
-            "1.4.13-3"
+            "1.4.13-3",
+            "1.6.4-4"
```
Comment (ganeshkumar5699) [inline parts/linux/cloud-init/artifacts/manifest.json:8]: Need to add all versions besides "latest" to this list of versions, as this is the list used for downloading (the "latest" version is downloaded and installed in the VHD by default).

File: pkg/agent/testdata/AKSUbuntu1604+CustomKubeletConfig+CustomLinuxOSConfig/line48.sh [hunk 2]
```diff
@@ -102,6 +102,7 @@ updateAptWithMicrosoftPkg() {
     fi
 
     retrycmd_if_failure 10 5 10 cp /tmp/microsoft-prod.list /etc/apt/sources.list.d/ || exit $ERR_MOBY_APT_LIST_TIMEOUT
+    echo "deb [arch=amd64,arm64,armhf] https://packages.microsoft.com/ubuntu/18.04/multiarch/prod testing main" > /etc/apt/sources.list.d/microsoft-prod-testing.list
```
Comment (ganeshkumar5699) [inline pkg/agent/testdata/AKSUbuntu1604+CustomKubeletConfig+CustomLinuxOSConfig/line48.sh:105]: Added the testing channel here. Thanks @cpuguy83 for the help with this.

File: parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh [hunk 3]
```diff
@@ -162,26 +163,28 @@ installStandaloneContainerd() {
...
 
 downloadContainerdFromVersion() {
-    #containerd has arm64 binaries from 1.4.4 or later on moby.blob.core.windows.net
-    CPU_ARCH=$(getCPUArch)  #amd64 or arm64
     CONTAINERD_VERSION=$1
-    # currently upstream maintains the package on a storage endpoint rather than an actual apt repo
-    CONTAINERD_PATCH_VERSION="${2:-1}"
-    CONTAINERD_DOWNLOAD_URL="https://moby.blob.core.windows.net/moby/moby-containerd/${CONTAINERD_VERSION}+azure/bionic/linux_${CPU_ARCH}/moby-containerd_${CONTAINERD_VERSION}+azure-ubuntu18.04u${CONTAINERD_PATCH_VERSION}_${CPU_ARCH}.deb"
-    downloadContainerdFromURL $CONTAINERD_DOWNLOAD_URL
+    updateAptWithMicrosoftPkg
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh:184]: is this not already done elsewhere?

**Review by alexeldeib — APPROVED** [review]
mostly lgtm, one minor improvement

File: parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh (hunk 3 above)
Comment (ganeshkumar5699) [reply parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh:184]: It is called in two places: 
1) In`downloadContainerdFromVersion` , which is called in during VHD build time and in CSE, in `installStandaloneContainerd`. 

2)  In`installMoby`, which should be called for docker builds. 

And I just reverted the earlier revert PR for this chang.

Comment (ganeshkumar5699) [reply parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh:184]: So yeah, this function would be called twice, if the necessary containerd version was not cached.

Comment (ganeshkumar5699) [reply parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh:184]: Perhaps a better way is to move `updateAptWithMicrosoftPkg` from the `downloadContainerdFromVersion` helper function, to being called once during the VHD build. Will do that.

File: parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh [hunk 4]
```diff
@@ -102,6 +102,7 @@ updateAptWithMicrosoftPkg() {
     fi
 
     retrycmd_if_failure 10 5 10 cp /tmp/microsoft-prod.list /etc/apt/sources.list.d/ || exit $ERR_MOBY_APT_LIST_TIMEOUT
+    echo "deb [arch=amd64,arm64,armhf] https://packages.microsoft.com/ubuntu/18.04/multiarch/prod testing main" > /etc/apt/sources.list.d/microsoft-prod-testing.list
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh:105]: please use UBUNTU_RELEASE here for 20.04, 22.04

Comment (alexeldeib) [reply parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh:105]: nvm, I see it doesn't exist for 20.04 and 22.04...something to perhaps follow up on

Comment (cpuguy83) [reply parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh:105]: Those repos don't have "multiarch" in the url.

Comment (ganeshkumar5699) [reply parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh:105]: @cpuguy83, thanks for the help with this! Probably a basic question: do the containerd builds for 20.04 and 22.04 differ, from the 18.04 builds? Wondering if it's ok to use the 18.04 URL for all (or change it depending on the version).

Comment (cpuguy83) [reply parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh:105]: @ganeshkumar5699 Yes, the builds are different as the binaries are dynamically linked against the toolchain available for each distro.

Comment (ganeshkumar5699) [reply parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh:105]: Thank you. Will likely be making this change in a subsequent PR.

**Pull request merged**